### DISCUSSION TITLE
Component selection for a transcription project

### DISCUSSION THREAD

**Discussion by Cinbert** [Budibase/budibase#1829] [Help] [answered]
My application requires me to display an image (attachment data type) from Table 1 and on the same screen,  list, and possibly create or edit transcription records from Table 2 that are related to the image being looked at. 

I have the basic search/filter set up and can display a table with a thumbnail of the images a row at a time (using row count) and a table with the transcriptions that meet the criteria in two containers below. 

There may be multiple images that meet the criteria and each one of them would have an associated set of transcription records.

Is there a component or combination that will allow me to display the actual image (not thumbnail) one at a time on the screen from Table 1 and the associated transcriptions from table 2 below?  I would like the user to be able to page through the images (table 1) that meet the criteria and enter transcriptions (table 2) for the one they were looking for.

I have tried many combinations of Forms, Repeaters and Cards and can't seem to find anything to fit the bill.  

Thanks for pointing me in the right direction.

**Comment by mjashanks** [comment, marked as answer]
Hi @Cinbert . Hope I've understood you correctly, check out my video below. It starts with a demo of what i've done, where I can display an image (saved in an `Images` table), with related transcriptions below it (stored in a `Transcriptions` table with a relationship to `Images`) 

So, I have
1. Created a data provider, with the datasource set to `Images`
2. I've set the limit of this data provider to 1, so that I can paginate one at a time. You can set this as you like, and add filters as you mention above.
3. Add a repeater to the data provider
4. Add an image to the repeater, and bind the URL property to the `Files` column. This is the attachment column that stores the image on the `Images` table
5. Add a Data Provider to the repeater, with data source set to the relationship between the two tables
6. Add a table to the new data provider

Let me know how it goes!

https://user-images.githubusercontent.com/3524181/123406062-6cb2af80-d5a2-11eb-9c32-f7dc7b01bd58.mp4

**Reply by Cinbert** [reply]
Michael, That was exactly what I needed and worked the first try.  It has helped me to understand a little better the nesting of components as well as the bindings.   Just one more question about the image display.  Is there a way to get the zoom and pan functionality from the browser like I would get if I right clicked and opened the image in a separate tab? Most of my images really require close inspection.  

 Thank you so much for going the extra step to implement and make the video. Your 6 steps were perfect too.

**Reply by mjashanks** [reply]
You're welcome!

We don't have anything inbuilt for this. Opening in a new tab seems like the best way at the moment.
